Teams review support

volas opened the pull request:
Adds support for team review:

```yaml
components:
  src/example.ts: org-name/team-name
```

But it does not work with standard ``github.token``, for some reason github needs more permissions for ``team_reviewers`` ("Could not resolve to a node with the global id of ..." error or something). I have updated README with steps for generating and using own token for this.

volas committed:
Teams review support

volas committed:
Fixed teams review re-request

dyladan commented:
@volas sorry to let this slip through the cracks. Is this something you're still interested in driving? I assume the `/` prefix is to disambiguate from usernames in text? What about `org-name/team-name` like github normally does? I know the org name will always be the same but I think reading it will be more obvious to most people.

volas committed:
Merge branch 'main' of https://github.com/dyladan/component-owners into teams_support

volas committed:
repo owner as team prefix

volas committed:
update readme with steps to use own access token

volas commented:
@dyladan no problem, I was using this on my fork and dropped this here just in case if someone else interested.

I have updated it to use "org-name/" as team prefix.

And also updated [Readme.md](https://github.com/dyladan/component-owners/blob/fe99bd3ef2a341467ffcf4b4e60fed9659479818/README.md) with instruction about using personal access token instead of `github.token`, because else action don't have permissions to request team review. This part can be little complicated.

svrnm commented on README.md:
```diff
@@ -98,6 +102,23 @@ Determines if the component owners should be added to the pull request as assign
+
+1. Just follow [Github's instructions](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#creating-a-fine-grained-personal-access-token) and add a new token with the following set of permissions:
+
+![perm](https://github.com/volas/component-owners/assets/4064061/474022d1-820a-48c6-8100-a8553584e961)
```
If this PR gets merged this should point to the upstream repo or have a link independent of the repo name

volas replied:
Replaced it with a markdown table.

Also removed Discussions prem as it was unnecessary, I think.

svrnm commented on README.md:
```diff
@@ -32,6 +32,9 @@ components:
   "*.md":
     - owner5
 
+  # You also can assign ownership for a team
+  src/example.ts: org-name/team-name
```
```suggestion
  # You also can assign ownership for a team (https://docs.github.com/en/organizations/organizing-members-into-teams/about-teams)
  src/example.ts: org-name/team-name
```

Maybe provide a link to what teams are if people lack that context

svrnm commented:
Thank you @volas! We are looking for the same functionality, so glad to see you already worked on that.

svrnm commented:
FYI, I did a small test run for my PR towards opentelemetry.io, it seems to work:

https://github.com/svrnm/opentelemetry.io/actions/runs/9791445867/job/27035250030

There have been no reviewers assigned since it checked for `open-telemetry/team-name` which of course is not available in my privat repo.

volas committed:
Permissions table instead of the screen

volas committed:
about teams link

Co-authored-by: Severin Neumann <neumanns@cisco.com>
